### ISSUE TITLE
The multi platform doesn't implement `Clone`

### ISSUE THREAD

**Issue by asajeffrey** [pcwalton/surfman#51]
Somewhat annoyingly, the multi platform derives `Clone`, but because of the definition:
```rust
#[derive(Clone)]
pub enum Connection<Def, Alt> where Def: DeviceInterface,
                                    Alt: DeviceInterface,
...
```
this implements `Clone` when `Def: Clone` and `Alt: Clone` but what's wanted is:
```rust
impl<Def, Alt> Clone for Connection<Def, Alt> where
    Def: DeviceInterface,
    Alt: DeviceInterface,
    Def::Connection: Clone,
    Alt::Connection: Clone,
```

**Comment by asajeffrey** [comment]
```
error[E0599]: no method named `clone` found for type `surfman::platform::generic::multi::connection::Connection<surfman::platform::unix::wayland::device::Device, surfman::platform::unix::x11::device::Device>` in the current scope
  --> /home/ajeffrey/github/asajeffrey/webxr/webxr/glwindow/mod.rs:96:46
   |
96 |             let connection = self.connection.clone();
   |                                              ^^^^^ method not found in `surfman::platform::generic::multi::connection::Connection<surfman::platform::unix::wayland::device::Device, surfman::platform::unix::x11::device::Device>`
   |
   = note: the method `clone` exists but the following trait bounds were not satisfied:
           `surfman::platform::generic::multi::connection::Connection<surfman::platform::unix::wayland::device::Device, surfman::platform::unix::x11::device::Device> : std::clone::Clone`
```

**Comment by asajeffrey** [comment]
Closed by #52
